components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java

```diff
@@ -293,23 +470,72 @@ public void testSendRequest() throws Exception {
         Assert.assertTrue(!result.isEmpty(), "The send request should not be empty.");
     }
 
+
+    @Test
+    public void testGetOauthResponseWithoutExceptions() throws Exception {
+        when(mockOAuthClient.accessToken(mockOAuthClientRequest)).thenReturn(mockOAuthJSONAccessTokenResponse);
+        openIDConnectAuthenticator.getOauthResponse(mockOAuthClient, mockOAuthClientRequest);
+    }
+
+
+    @Test(expectedExceptions = AuthenticationFailedException.class)
+    public void testGetOauthResponseWithExceptions() throws Exception {
+        OAuthClientRequest oAuthClientRequest = mock(OAuthClientRequest.class);
+        OAuthClient oAuthClient = mock(OAuthClient.class);
+        when(oAuthClient.accessToken(oAuthClientRequest)).thenThrow(OAuthSystemException.class);
+        setMockedLog(true);
+        openIDConnectAuthenticator.getOauthResponse(oAuthClient, oAuthClientRequest);
+    }
+
+    @Test
+    public void testGetOauthResponseWithOAuthProblemExceptions() throws Exception {
```

Comment:
No Asserts or Expected Exceptions?

Reply:
Also why did we throw a generic exception?

Reply:
done